src/vs/workbench/contrib/debug/node/debugAdapter.ts

```diff
@@ -222,13 +222,22 @@ export class ExecutableDebugAdapter extends StreamDebugAdapter {
 					throw new Error(nls.localize('unableToLaunchDebugAdapterNoArgs', "Unable to launch debug adapter."));
 				}
 			} else {
+				let spawnCommand = command;
+				let spawnArgs = args;
 				const spawnOptions: cp.SpawnOptions = {
 					env: env
 				};
 				if (options.cwd) {
 					spawnOptions.cwd = options.cwd;
 				}
-				this.serverProcess = cp.spawn(command, args, spawnOptions);
+				if (platform.isWindows && command.endsWith('.bat')) {
+					// https://github.com/microsoft/vscode/issues/224184
+					spawnOptions.shell = true;
+					spawnCommand = `"${command}"`;
+					spawnArgs = args.map(a => `"${a}"`);
```

Comment:
If any args contain `"` do those need escaping?

(mine don't, and hopefully others don't, but who knows?)

Reply:
Looks like they do 😞

Reply:
IIRC things like "<" can cause issues too (see https://ss64.com/nt/syntax-esc.html). In Dart-Code I have this escape function:

https://github.com/Dart-Code/Dart-Code/blob/d1894e79c99360438d509922e5b8ff391ecdbcf0/src/shared/processes.ts#L29

I don't know if it's reliable (particularly if you changed `COMSPEC` to PoSh, but hopefully nobody is doing that 😄) or how likely it is in the debug adapter executable (it came up for me because [test names can be passed as args](https://github.com/Dart-Code/Dart-Code/issues/4042) to some processes and that's a shared escape function).

Reply:
Are you already escaping any args that go to the DebugAdapterExecutable? Or is it a fixed set of args?

Reply:
No, not escaping anything for `DebugAdapterExecutable` because it's not usually required (the function above is used in the places where we are using `shell:true` to spawn our own processes).

> Or is it a fixed set of args?

Yep, for the place that will come through `DebugAdapterExecutable` our args are fixed (well, they're dynamic, but from a small set which is basically `["debug_adapter", "--test", "--no-dds"]`). There are a few small exceptions (where you might pass a full path), but they're generally just for me (or other contributors to the debug adapter) that are running from source.

(TL;DR; simple escaping such as only putting quotes around them is definitely good enough for me, and the comments above were just general comments about being strict in case others are doing weirder things)

Reply:
This level of escaping seems to be fine, including for characters like `>`, although cmd.exe escaping is obviously extremely weird.

Did you test that this works for your scenarios @DanTup?